pyEpiabm/pyEpiabm/tests/test_unit/test_routine/test_file_population_config.py

```diff
@@ -194,8 +197,17 @@ def test_print_population(self, mock_copy, mock_read):
 
         FilePopulationFactory.print_population(test_pop, 'output.csv')
         if version.parse(pd.__version__) >= version.parse("1.4.0"):
-            pd.testing.assert_frame_equal(mock_copy.call_args.args[0],
-                                          self.df, check_dtype=False)
+            expected_df = self.df.copy()
+            expected_df.drop('household_number', axis=1)
+
+            actual_df = mock_copy.call_args.args[0]
+
+            pd.testing.assert_frame_equal(actual_df,
+                                          expected_df, check_dtype=False)
+            household_list = self.df.loc[:, 'household_number']
+            actual_household = actual_df.loc[:, 'household_number']
+            for i in range(len(household_list)):
+                self.assertLessEqual(actual_household[i], household_list[i])
```

Comment:
Why is `assertLessEqual` used here, should the values not just be the same? Also this check can be outside the if clause, as it doesn't require the updated pandas that line 205 does

Reply:
Because empty households are no longer allowed the number of households after configuring the population may be lower than the number initial specified, hence LessEqual.